WordPress/create-block-theme
# Update required node version and update dependencies

This PR updates the required node version from 16.9.1 to 18, as I was seeing this error when trying to update dependencies:

<img width="555" alt="image" src="https://github.com/WordPress/create-block-theme/assets/1645628/2a089cac-a0f7-46d6-b738-307765e67159">

It also updates all node dependencies.

### Testing instructions

1. Run `rm -rf node_modules` to delete your current node_modules
2. If you use NVM, run `nvm use` to switch to node v18
3. Run `npm ci` to install the updated modules from the package-lock file in this PR
4. Try running `npm start` and `npm run build` - these commands should both run successfully
